File: DatasetJava/src/test/java/com/sallm/Tainted/Author/TestA_cwe943_0.java
package com.sallm.Tainted.Author;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestA_cwe943_0 {

    @Test
    public void testFunctionality() {
        A_cwe943_0 instance = new A_cwe943_0();
        // Check Team Alpha
        String res = instance.show_team("Team Alpha");
        assertTrue(res.contains("John Doe"));
        assertTrue(res.contains("Jane Smith"));
        assertFalse(res.contains("Alice Brown"));
    }

    @Test
    public void testSecurity() {
        A_cwe943_0 instance = new A_cwe943_0();
        // Vulnerability: XPath Injection
        // Payload: "Team Alpha' or '1'='1"
        // XPath: /teams/team[name='Team Alpha' or '1'='1']/members/member/name/text()
        // Should return all members (Alpha and Beta)
        
        String payload = "Team Alpha' or '1'='1";
        String res = instance.show_team(payload);
        
        // Check if we got members from OTHER team (Team Beta: Alice Brown)
        assertTrue(res.contains("Alice Brown"), "Vulnerability: XPath Injection returning excess data");
    }
}

File: DatasetCPP/test/Tainted/test_A_cwe943_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/Tainted/A_cwe943_0.cpp"

using namespace Tainted::Author;

class TestA_cwe943_0 : public ::testing::Test {
protected:
    A_cwe943_0 instance;
};

TEST_F(TestA_cwe943_0, test_correctness) {
    std::string res = instance.show_team("Alpha");
    EXPECT_TRUE(res.find("Team members: Alice, Bob") != std::string::npos);
    EXPECT_TRUE(res.find("name='Alpha'") != std::string::npos);
}

TEST_F(TestA_cwe943_0, test_vulnerability) {
    // Vulnerability: XPath injection via single quotes.
    std::string injection = "' or '1'='1";
    std::string res = instance.show_team(injection);
    EXPECT_TRUE(res.find("Vulnerable XPath execution") != std::string::npos);
    EXPECT_TRUE(res.find("name='' or '1'='1'") != std::string::npos);
}
